Assets/Scripts/Player/PlayerShooting.cs

```diff
@@ -11,18 +11,37 @@ public sealed class PlayerShooting : MonoBehaviour
 
     private GameObject _bullet;
     private float _shootingPower;
+    private float _shootingDelay;
+    private float _currentShootDelay;
+    private bool _autoShooting;
 
-    private void Awake() => SetGunData(startingGunData);
+    private void Awake()
+    {
+        SetGunData(startingGunData);
+        _currentShootDelay = startingGunData.shootingDelay;
+    }
 
     private void Update()
     {
-        if(Input.GetButtonDown("Fire1")) Shoot();
+        _currentShootDelay -= Time.deltaTime;
+        if (_autoShooting && Input.GetButton("Fire1")) DelayedShooting();
+        else if (Input.GetButtonDown("Fire1")) DelayedShooting();
     }
 
     public void SetGunData(GunData targetData)
```

Comment:
```suggestion
    public void SetGunData(GunData gunData)
```
De naam `targetData` suggereert dat deze parameter aangepast gaat worden vanwege het woord "target". Aangezien de method ook al `SetGunData` heet zie ik niet per sé de toegevoegde waarde om de naam van deze parameter abstract te houden.